### ISSUE TITLE
安桌shell里运行linux可执行程序失败怎么知道缺失了哪些.so??这玩意不显示的

### ISSUE THREAD

**Issue by expnkx** [mzlogin/awesome-adb#73]
![Screenshot from 2019-12-31 14-49-47](https://user-images.githubusercontent.com/57303909/71632488-d257f280-2bdc-11ea-9100-7e9956efec94.png)
根本不显示。即使可执行程序在这里。

**Comment by mzlogin** [comment]
具体我没有研究过，但 Linux 可执行程序应该不是那么容易拷进去就可用的，应该需要专门的移植适配版，可以使用像 busybox for android 之类的。

你图里的这个问题像是 32 位 64 位的问题。

**Comment by expnkx** [comment]
> 具体我没有研究过，但 Linux 可执行程序应该不是那么容易拷进去就可用的，应该需要专门的移植适配版，可以使用像 busybox for android 之类的。
> 
> 你图里的这个问题像是 32 位 64 位的问题。

不是。我的实验表明如果是静态编译的可执行文件就能用。动态链接的就用不了，但由于安卓shell不告诉我哪些so缺失，导致没法折腾这点上

**Comment by mzlogin** [comment]
你可以试试用 ldd 命令先在 Linux 下看查询它依赖哪些 so 文件，然后都拷进去试试。

**Comment by canyie** [comment]
试一下编译时加上-fPIE

**Comment by expnkx** [comment]
> 试一下编译时加上-fPIE

-fPIE有什么用呢？

**Comment by canyie** [comment]
安卓5.0以后强制要求可执行文件以-pie编译，否则会报错
error: only position independent executables (PIE) are supported.
或许是这个原因？

**Comment by expnkx** [comment]
> 安卓5.0以后强制要求可执行文件以-pie编译，否则会报错

还是不行。我传了libc.so.6根本没用

**Comment by expnkx** [comment]
> 安卓5.0以后强制要求可执行文件以-pie编译，否则会报错
> error: only position independent executables (PIE) are supported.
> 或许是这个原因？

libc.so.6都没用，什么都折腾不了实际上
